src/main/kotlin/io/openfuture/snapshot/repository/ArchivedNodeInMemorySnapshotRepository.kt

```diff
@@ -0,0 +1,20 @@
+package io.openfuture.snapshot.repository
+
+import io.openfuture.snapshot.dto.ArchivedNodeSnapshotDto
+import io.openfuture.snapshot.dto.SnapshotDto
+
+class ArchivedNodeInMemorySnapshotRepository : InMemorySnapshotRepository() {
+
+    override fun saveAll(snapshots: List<SnapshotDto>) {
+        snapshots.forEach {
+            it as ArchivedNodeSnapshotDto
+            val value = snapshotsMap[it.address]
+            if (null != value) {
+                snapshotsMap.replace(it.address, it.balance)
+            } else {
```

Comment:
put would override value is key already exist and add new node if key not found. No need such logic